src/controller/SetUpCodePairer.cpp

```diff
@@ -310,7 +316,45 @@ bool SetUpCodePairer::ConnectToDiscoveredDevice()
         SetUpCodePairerParameters params(mDiscoveredParameters.front());
         mDiscoveredParameters.pop_front();
 
-        params.SetSetupPINCode(mSetUpPINCode);
+        if (params.mLongDiscriminator)
+        {
+            auto longDiscriminator = *params.mLongDiscriminator;
+            // Look for a matching setup passcode.
+            bool found = false;
+            for (auto & payload : mSetupPayloads)
+            {
+                if (payload.discriminator.MatchesLongDiscriminator(longDiscriminator))
+                {
+                    params.SetSetupPINCode(payload.setUpPINCode);
+                    found = true;
+                    break;
+                }
+            }
+            if (!found)
+            {
+                ChipLogError(Controller, "SetUpCodePairer discovered discriminator %u does not match any of our setup payloads",
```

Comment:
```suggestion
                ChipLogError(Controller, "SetUpCodePairer: Discovered discriminator %u does not match any of our setup payloads",
```

(Or get rid of the "SetupCodePairer" prefix entirely since the rest of the logging in this class doesn't appear to do that either.)